### PR TITLE
Update deploy translatable strings script to push commit hash

### PR DESCRIPTION
**Pull request by SooluThomas**
### Summary

this is to pull the correct commit that is not main _(which is how translations repo pull in qiskit repo for docs build)_ or to not wait till the actual release

### CODE REVIEW

File: tools/deploy_translatable_strings.sh [hunk 1]
```diff
@@ -89,6 +89,8 @@ cp "$SOURCE_REPO_ROOT/qiskit_pkg/setup.py" "${TARGET_REPO_ROOT}"
 cat "$SOURCE_REPO_ROOT/requirements-dev.txt" "$SOURCE_REPO_ROOT/requirements-optional.txt" \
     > "${TARGET_REPO_ROOT}/requirements-dev.txt"
 cp "$SOURCE_REPO_ROOT/constraints.txt" "${TARGET_REPO_ROOT}"
+# Add commit hash to be able to run the build with the commit hash before the actual release
+echo $QISKIT_COMMIT > "${TARGET_REPO_ROOT}/qiskit-commit-hash"
```
Comment (jakelishman) [inline tools/deploy_translatable_strings.sh]: As it stands, we haven't actually defined this `QISKIT_COMMIT` environment variable yet in this file.  We'll need to either ask `git` for it (from the source repo), or feed it into this file from the CI pipelines.

Comment (SooluThomas) [reply tools/deploy_translatable_strings.sh]: Oh I think it should be `GITHUB_SHA` instead

File: tools/deploy_translatable_strings.sh [hunk 2]
```diff
@@ -89,6 +89,8 @@ cp "$SOURCE_REPO_ROOT/qiskit_pkg/setup.py" "${TARGET_REPO_ROOT}"
 cat "$SOURCE_REPO_ROOT/requirements-dev.txt" "$SOURCE_REPO_ROOT/requirements-optional.txt" \
     > "${TARGET_REPO_ROOT}/requirements-dev.txt"
 cp "$SOURCE_REPO_ROOT/constraints.txt" "${TARGET_REPO_ROOT}"
+# Add commit hash to be able to run the build with the commit hash before the actual release
+echo $GITHUB_SHA > "${TARGET_REPO_ROOT}/qiskit-commit-hash"
 
 echo "+ 'git add' files to target commit"
 git add docs/ setup.py requirements-dev.txt constraints.txt
```
Comment (jakelishman) [inline tools/deploy_translatable_strings.sh]:
```suggestion
git add docs/ setup.py requirements-dev.txt constraints.txt qiskit-commit-hash
```

Need to make sure that we've added the file to the commit as well - sorry, I missed that the first time.  Other than this comment, this looks good to merge to me.

Comment (SooluThomas) [reply tools/deploy_translatable_strings.sh]: Yes! I missed that too 😄 Thank you for catching that

**Review by jakelishman — APPROVED** [review]
Thanks Soolu, this looks good to me.  I can set this working through Terra's backport machinery (at least a couple of hours) while the companion PR gets prepared for the translations repository.

**Comment by jakelishman** [comment]
@Mergifyio backport stable/0.25

**Comment by jakelishman** [comment]
Oh wait, I hadn't even realised that you'd made this PR directly to `stable/0.25`.  We actually need it on `main` and `stable/0.45` as well, so I'll just _forward_ port it to those.

@Mergifyio backport stable/0.45 main

**Comment by jakelishman** [comment]
@Mergifyio backport stable/0.45 main

**Comment by SooluThomas** [comment]
Ah, I didn't realize that I was on stable/0.25 when I committed the changes. Sorry

**Comment by jakelishman** [comment]
No worries, it's a good excuse for me to learn if the bot will "backport" between completely arbitrary branches!

**Pull request merged**